src/consensus/aft/raft.h

```diff
@@ -243,6 +245,12 @@ namespace aft
         // Initialize view history for bft. We start on view 2 and the first
         // commit is always 1.
         state->view_history.update(1, starting_view_change);
+        use_two_tx_reconfig = true;
+      }
+
+      if (use_two_tx_reconfig && !public_only)
```

Comment:
That doesn't seem right to me, whether the node thinks it is in public_only mode or not should have no bearing on whether it becomes learner. It's to do with whether the service is open or not already, regardless of whether it is in recovery or startup mode.

Reply:
That's what I thought too. It seems the `public_only` flag is only used for recovery, but I may be wrong about that. Are there other flags that let us determine whether we should be starting as a learner or follower?

Reply:
It is only used for recovery, but recovery is not what decides if the node can skip learner or not. It's whether the network is pre-OPEN, which can happen both at startup and in recovery. The return of the join RPC from the network is what tells the node what state it can enter (currently either PENDING or TRUSTED).

Reply:
-> join RPC will tell us